shapely/geometry/base.py

```diff
@@ -57,18 +57,20 @@ def dump_coords(geom):
         # Recursive call
         return [dump_coords(part) for part in geom.geoms]
     else:
-        raise ValueError('Unhandled geometry type: ' + repr(geom.type))
+        raise GeometryTypeError('Unhandled geometry type: ' + repr(geom.type))
 
 
 def geometry_type_name(g):
     if g is None:
+        # Do we consider this a type error?
         raise ValueError("Null geometry has no type")
     return GEOMETRY_TYPES[lgeos.GEOSGeomTypeId(g)]
 
 
 def geom_factory(g, parent=None):
     # Abstract geometry factory for use with topological methods below
     if not g:
+        # Do we consider this a type error?
```

Comment:
This is not a type error. Also, geom_factory(), while not hidden from users, is for internal use and not part of the official public API. We can leave the ValueError here.